worlds/witness/__init__.py

```diff
@@ -40,39 +41,52 @@ class WitnessWorld(World):
     topology_present = False
     data_version = 13
 
-    static_logic = StaticWitnessLogic()
-    static_locat = StaticWitnessLocations()
-    static_items = StaticWitnessItems()
+    StaticWitnessLogic()
+    StaticWitnessLocations()
+    StaticWitnessItems()
     web = WitnessWebWorld()
     option_definitions = the_witness_options
 
     item_name_to_id = {
-        name: data.code for name, data in static_items.ALL_ITEM_TABLE.items()
+        name: data.ap_code for name, data in StaticWitnessItems.item_data.items()
     }
     location_name_to_id = StaticWitnessLocations.ALL_LOCATIONS_TO_ID
-    item_name_groups = StaticWitnessItems.ITEM_NAME_GROUPS
+    item_name_groups = StaticWitnessItems.item_groups
 
     required_client_version = (0, 3, 9)
 
+    def __init__(self, multiworld: game, player: int):
+        super().__init__(multiworld, player)
+
+        disabled_locations = self.multiworld.exclude_locations[self.player].value
+
+        self.player_logic = WitnessPlayerLogic(
+            self.multiworld, self.player, disabled_locations, self.multiworld.start_inventory[self.player].value
+        )
+
+        self.locat: WitnessPlayerLocations = WitnessPlayerLocations(self.multiworld, self.player, self.player_logic)
+        self.items: WitnessPlayerItems = WitnessPlayerItems(self.multiworld, self.player, self.player_logic, self.locat)
+        self.regio: WitnessRegions = WitnessRegions(self.locat)
+
+        self.log_ids_to_hints = dict()
+
```

Comment:
Did you test all of these things? I was not actually under the impression that all options, including exclude_locations, was loaded by the time of __init__.py. If it is, that's fine

Reply:
Ah yeah. This has, in fact, created a unit test error.

Reply:
Makes sense, I'll move it back. I put these here since you're not supposed to add new object members outside of `__init__`, but I can just initialize them to None and fill them out in `generate_early`.